### PR TITLE
[msal-node-extensions] Add persistence creator

### PR DESCRIPTION
**Pull request by samuelkubai**
- add a factory function for the persistence instance
- detect the right environment for the right persistence instance
- provide a `getUserRootDirectory` function as part of our API to aid in cache file path creation

### CODE REVIEW

File: change/@azure-msal-node-extensions-0bdf3389-5c5a-4ab3-8520-ff2adb9fb815.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
Comment (jasonnutter) [inline change/@azure-msal-node-extensions-0bdf3389-5c5a-4ab3-8520-ff2adb9fb815.json:2]:
```suggestion
  "type": "prerelease",
```

Needs to be prereleases which extensions is still in alpha/beta

Comment (samuelkubai) [reply change/@azure-msal-node-extensions-0bdf3389-5c5a-4ab3-8520-ff2adb9fb815.json:2]: My bad, keeps slipping my mind.

**Review by jasonnutter — APPROVED** [review]
Looks good!

File: change/@azure-msal-node-extensions-0bdf3389-5c5a-4ab3-8520-ff2adb9fb815.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "feat: add the persistence creator",
```
Comment (sameerag) [inline change/@azure-msal-node-extensions-0bdf3389-5c5a-4ab3-8520-ff2adb9fb815.json:3]:
```suggestion
  "comment": "feat: add the persistence creator (#3859)",
```

File: extensions/msal-node-extensions/src/error/PersistenceError.ts [hunk 3]
```diff
@@ -68,4 +68,34 @@ export class PersistenceError extends Error {
...
+        return new PersistenceError("PersistenceNotVerified", errorMessage);
+    }
+
+    /**
+     * Throw persistence creation validation error
+     * 
+     * @param errorMessage string
+     * @returns PersistenceError
+     */
+    static createPersistenceCreationValidationError(errorMessage: string): PersistenceError {
```
Comment (sameerag) [inline extensions/msal-node-extensions/src/error/PersistenceError.ts:98]: nit: `PersistenceCreationValidation` is confusing. Any better names?

**Review by sameerag — APPROVED** [review]
Some minor comments, approved.

**Pull request merged**